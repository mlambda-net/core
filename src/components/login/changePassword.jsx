import React from 'react';
import { Box, Button, FormControl, Paper, TextField } from '@mui/material';
import LocalizedStrings from 'react-localization';
import { withUtils } from '@mlambda-net/web-core/utils';
import { Title } from '@mlambda-net/web-core/common';

const language = new LocalizedStrings({
  en: {
    title: 'Change Password',
    newPassword: 'New Password',
    oldPassword: 'Old Password',
    email: 'Email',
    errorNewPassword: 'new password is required',
    errorOldPassword: 'old password is required',
    changeAction: 'submit',
  },
  es: {
    title: 'Cambiar contraseña',
    newPassword: 'Contraseña Nueva',
    oldPassword: 'Contraseña Actual',
    email: 'Email',
    errorNewPassword: 'la contraseña nueva es requerida',
    errorOldPassword: 'la contraseña actual es requerida',
    changeAction: 'cambiar',
  },
});

const styles = (theme) => ({});

class ChangePassword extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      newPassword: '',
      oldPassword: '',
      validOldPassword: false,
      validNewPassword: false,
      errorNewPassword: '',
      errorOldPassword: '',
    };

    this.valid = this.props.validation;
    this.newPasswordChange = this.newPasswordHandle.bind(this);
    this.oldPasswordChange = this.oldPasswordHandle.bind(this);
    this.changeClick = this.changeHandle.bind(this);
  }

  render() {
    language.setLanguage(this.props.lang);
    const { classes } = this.props;
    return (
      <Paper elevation={10} className={this.props.className}>
        <Box p={3}>
          <Box display="flex" justifyContent="center">
            <Title title={language.title} />
          </Box>
          <Box
            display="flex"
            flexDirection="column"
            justifyContent="center"
            m={1}
            p={1}>
            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="password"
                  label={language.oldPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validOldPassword}
                  onChange={this.oldPasswordChange}
                  helperText={this.state.errorOldPassword}
                />
              </FormControl>
            </Box>

            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="confirm_password"
                  label={language.newPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validNewPassword}
                  onChange={this.newPasswordChange}
                  helperText={this.state.errorNewPassword}
                />
              </FormControl>
            </Box>

            <Box p={1} m={1} display="flex" justifyContent="flex-end">
              <Button
                variant="outlined"
                color="primary"
                disabled={!this.isValid()}
                onClick={this.changeClick}>
                {language.changeAction}
              </Button>
            </Box>
          </Box>
        </Box>
      </Paper>
    );
  }

  newPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorNewPassword: language.errorNewPassword,
      });
    } else {
      this.setState({
        newPassword: password,
        validNewPassword: true,
        errorNewPassword: '',
      });
    }
  }

  oldPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorOldPassword: language.errorOldPassword,
      });
    } else {
      this.setState({
        oldPassword: password,
        validOldPassword: true,
        errorOldPassword: '',
      });
    }
  }

  changeHandle(event) {
    this.props.onChange({
      newPassword: this.state.newPassword,
      oldPassword: this.state.oldPassword,
    });

    event.preventDefault();
  }

  isValid() {
    return this.state.validNewPassword && this.state.validOldPassword;
  }
}

export default withUtils(styles)(ChangePassword);
