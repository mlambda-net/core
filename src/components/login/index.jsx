export { default as Register } from './register';
export { default as SignIn } from './signin';
export { default as ChangePassword } from './changePassword';
export { default as Recovery } from './recovery';
export { default as Profile } from './profile';
