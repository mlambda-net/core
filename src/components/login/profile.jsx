import React from 'react';
import LocalizedStrings from 'react-localization';
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import { Box, Card, CardActions, CardContent, Typography } from '@mui/material';
import { Option } from '@mlambda-net/web-core/common';
import { withUtils } from '@mlambda-net/web-core/utils';

const language = new LocalizedStrings({
  en: {
    title: 'Profile',
  },
  es: {
    title: 'Perfil',
  },
});

const styles = (theme) => ({
  panel: {
    width: '100%',
    height: '100%',
  },

  title: {
    textAlign: 'center',
  },
  content: {
    height: 'calc(100% - 80px)',
  },
  actions: {
    display: 'flex',
    flexWrap: 'nowrap',
    justifyContent: 'flex-end',
  },
});

class Profile extends React.Component {
  constructor(props) {
    super(props);
    this.valid = this.validation;
    this.state = { open: false };
  }

  render() {
    language.setLanguage(this.props.lang);
    const Content = this.props.content;
    const Actions = this.props.actions;
    const { classes } = this.props;

    return (
      <Option
        icon={AccountCircleIcon}
        width={this.props.width}
        height={this.props.height}
        style={this.props.style}>
        <Card className={classes.panel}>
          <CardContent className={classes.content}>
            <Typography
              className={classes.title}
              color="textSecondary"
              variant="h5"
              gutterBottom>
              {language.title}
            </Typography>
            <Content user={this.props.user} />
          </CardContent>

          <CardActions className={classes.actions}>
            <Box>
              <Actions />
            </Box>
          </CardActions>
        </Card>
      </Option>
    );
  }
}

export default withUtils(styles)(Profile);
