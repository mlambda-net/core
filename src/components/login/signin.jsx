import React from 'react';
import LocalizedStrings from 'react-localization';
import { withUtils } from '@mlambda-net/web-core/utils';
import {
  Box,
  Button,
  FormControl,
  Link,
  TextField,
  Typography,
} from '@mui/material';

const language = new LocalizedStrings({
  en: {
    password: 'Password',
    email: 'Email',
    invalidEmail: 'invalid email',
    invalidPassword: 'password is required',
    forgot: 'forgot password?',
    login: 'login',
  },
  es: {
    password: 'Contraseña',
    email: 'Correo',
    invalidEmail: 'correo invalido',
    invalidPassword: 'una contraseña es requerida',
    forgot: 'olvido la contraseña?',
    login: 'ingresar',
  },
});

const styles = (theme) => ({});

class SignIn extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      validEmail: false,
      validPassword: false,
      helpEmail: '',
      helpPassword: '',
    };
    this.valid = this.props.validation;
    this.emailChange = this.emailChangeHandle.bind(this);
    this.passwordChange = this.passwordChangeHandle.bind(this);
    this.loginClick = this.loginHandle.bind(this);
  }

  render() {
    language.setLanguage(this.props.lang);

    return (
      <Box width="100%">
        <Box display="flex" justifyContent="center">
          <Typography color="primary" variant="h4" component="h5" gutterBottom>
            {this.props.title}
          </Typography>
        </Box>
        <Box
          display="flex"
          flexDirection="column"
          justifyContent="center"
          m={1}
          p={1}>
          <Box p={1}>
            <FormControl fullWidth>
              <TextField
                label={language.email}
                color="secondary"
                id="email"
                type="email"
                error={!this.state.validEmail}
                onChange={this.emailChange}
                helperText={this.state.helpEmail}
              />
            </FormControl>
          </Box>
          <Box p={1}>
            <FormControl fullWidth>
              <TextField
                id="password"
                label={language.password}
                color="secondary"
                type="password"
                error={!this.state.validPassword}
                onChange={this.passwordChange}
                helperText={this.state.helpPassword}
              />
            </FormControl>
          </Box>

          <Box p={1} m={2} display="flex" justifyContent="flex-end">
            <Link color="secondary" onClick={this.props.onForgot}>
              <Typography variant="caption">{language.forgot}</Typography>
            </Link>
          </Box>

          <Box p={1} m={1} display="flex" justifyContent="flex-end">
            <Button
              variant="outlined"
              color="primary"
              disabled={!this.isValid()}
              onClick={this.loginClick}>
              {language.login}
            </Button>
          </Box>
        </Box>
      </Box>
    );
  }

  emailChangeHandle(event) {
    if (this.valid.email(event.target.value)) {
      this.setState({
        validEmail: true,
        helpEmail: '',
        email: event.target.value,
      });
    } else {
      this.setState({ validEmail: false, helpEmail: language.invalidEmail });
    }
  }

  passwordChangeHandle(event) {
    if (!this.valid.isEmpty(event.target.value)) {
      this.setState({
        validPassword: true,
        helpPassword: '',
        password: event.target.value,
      });
    } else {
      this.setState({
        validPassword: false,
        helpPassword: language.invalidPassword,
      });
    }
  }

  loginHandle(event) {
    this.props.onLogin({
      email: this.state.email,
      password: this.state.password,
    });
    event.preventDefault();
  }

  isValid() {
    return this.state.validEmail && this.state.validPassword;
  }
}

export default withUtils(styles)(SignIn);
