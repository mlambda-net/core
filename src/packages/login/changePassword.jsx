import React from 'react';
import Paper from '@material-ui/core/Paper';
import FormControl from '@material-ui/core/FormControl';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import LocalizedStrings from 'react-localization';
import { withUtils } from '@mlambda-net/core/utils';
import { Title } from '@mlambda-net/core/common';

const language = new LocalizedStrings({
  en: {
    title: 'Change Password',
    newPassword: 'New Password',
    oldPassword: 'Old Password',
    email: 'Email',
    errorNewPassword: 'new password is required',
    errorOldPassword: 'old password is required',
    changeAction: 'submit',
  },
  es: {
    title: 'Cambiar contraseña',
    newPassword: 'Contraseña Nueva',
    oldPassword: 'Contraseña Actual',
    email: 'Email',
    errorNewPassword: 'la contraseña nueva es requerida',
    errorOldPassword: 'la contraseña actual es requerida',
    changeAction: 'cambiar',
  },
});

const styles = (theme) => ({});

class ChangePassword extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      newPassword: '',
      oldPassword: '',
      validOldPassword: false,
      validNewPassword: false,
      errorNewPassword: '',
      errorOldPassword: '',
    };

    this.valid = this.props.validation;
    this.newPasswordChange = this.newPasswordHandle.bind(this);
    this.oldPasswordChange = this.oldPasswordHandle.bind(this);
    this.changeClick = this.changeHandle.bind(this);
  }

  render() {
    language.setLanguage(this.props.lang);
    const { classes } = this.props;
    return (
      <Paper elevation={10} className={this.props.className}>
        <Box p={3}>
          <Box display="flex" justifyContent="center">
            <Title title={language.title} />
          </Box>
          <Box
            display="flex"
            flexDirection="column"
            justifyContent="center"
            m={1}
            p={1}>
            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="password"
                  label={language.oldPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validOldPassword}
                  onChange={this.oldPasswordChange}
                  helperText={this.state.errorOldPassword}
                />
              </FormControl>
            </Box>

            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="confirm_password"
                  label={language.newPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validNewPassword}
                  onChange={this.newPasswordChange}
                  helperText={this.state.errorNewPassword}
                />
              </FormControl>
            </Box>

            <Box p={1} m={1} display="flex" justifyContent="flex-end">
              <Button
                variant="outlined"
                color="primary"
                disabled={!this.isValid()}
                onClick={this.changeClick}>
                {language.changeAction}
              </Button>
            </Box>
          </Box>
        </Box>
      </Paper>
    );
  }

  newPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorNewPassword: language.errorNewPassword,
      });
    } else {
      this.setState({
        newPassword: password,
        validNewPassword: true,
        errorNewPassword: '',
      });
    }
  }

  oldPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorOldPassword: language.errorOldPassword,
      });
    } else {
      this.setState({
        oldPassword: password,
        validOldPassword: true,
        errorOldPassword: '',
      });
    }
  }

  changeHandle(event) {
    this.props.onChange({
      newPassword: this.state.newPassword,
      oldPassword: this.state.oldPassword,
    });

    event.preventDefault();
  }

  isValid() {
    return this.state.validNewPassword && this.state.validOldPassword;
  }
}

ChangePassword.protoTypes = {
  className: PropTypes.string,
  classes: PropTypes.object.isRequired,
  onChange: PropTypes.func,
  lang: PropTypes.string,
};

export default withUtils(styles)(ChangePassword);
