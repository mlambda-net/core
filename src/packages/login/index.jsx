export Register from './register';
export SignIn from './signin';
export Profile from './profile'
export ChangePassword from './changePassword'
export Recovery from './recovery'
