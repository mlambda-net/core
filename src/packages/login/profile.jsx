import React from 'react';
import PropTypes from 'prop-types';
import LocalizedStrings from 'react-localization';
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import CardActions from '@mui/material/CardActions';
import { Box } from '@mui/material';
import { Option } from '@mlambda-net/core/common';
import withUtils from '@mlambda-net/core/utils/withUtils';

const language = new LocalizedStrings({
  en: {
    title: 'Profile',
  },
  es: {
    title: 'Perfil',
  },
});

const styles = (theme) => ({
  panel: {
    width: '100%',
    height: '100%',
  },

  title: {
    textAlign: 'center',
  },
  content: {
    height: 'calc(100% - 80px)',
  },
  actions: {
    display: 'flex',
    flexWrap: 'nowrap',
    justifyContent: 'flex-end',
  },
});

class Profile extends React.Component {
  constructor(props) {
    super(props);
    this.valid = this.validation;
    this.state = { open: false };
  }

  render() {
    language.setLanguage(this.props.lang);

    const Content = this.props.content;
    const Actions = this.props.actions;
    const { classes } = this.props;

    return (
      <Option
        icon={AccountCircleIcon}
        width={this.props.width}
        height={this.props.height}
        style={this.props.style}>
        <Card className={classes.panel}>
          <CardContent className={classes.content}>
            <Typography
              className={classes.title}
              color="textSecondary"
              variant="h5"
              gutterBottom>
              {language.title}
            </Typography>
            <Content />
          </CardContent>

          <CardActions className={classes.actions}>
            <Box>
              <Actions />
            </Box>
          </CardActions>
        </Card>
      </Option>
    );
  }
}

Profile.protoTypes = {
  className: '',
  classes: PropTypes.object.isRequired,
  user: PropTypes.object,
  lang: PropTypes.string,
  content: PropTypes.element,
  actions: PropTypes.element,
  width: PropTypes.string,
  height: PropTypes.string,
  style: PropTypes.string,
};

export default withUtils(styles)(Profile);
