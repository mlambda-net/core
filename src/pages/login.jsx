import React from 'react';

import { withStyles, withTheme } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import login from '../assets/img/login.png';
import SignIn from '@mlambda-net/core/login/signin';

const styles = (themes) => ({
  root: {
    height: '100%',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
  },
  login: {
    width: '600px',
  },
});

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.onLogin = this.onLoginHandle.bind(this);
  }

  render() {
    const { classes } = this.props;

    return (
      <Box
        display="flex"
        justifyContent="center"
        alignItems="center"
        className={classes.root}>
        <SignIn
          className={classes.login}
          icon={<img src={login} width="100%" />}
          onLogin={this.onLogin}
        />
      </Box>
    );
  }

  onLoginHandle(data) {
    console.log(data);
  }
}

export default withStyles(styles)(withTheme(Login));
