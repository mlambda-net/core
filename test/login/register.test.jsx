import React from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from 'vitest';
import { Register } from '@mlambda-net/web-core/login';
import Wrap from '../util';

test('Register Test', () => {
  const item = renderer.create(
    <Wrap>
      <Register onRegister={() => console.log('login')} />
    </Wrap>,
  );
  let tree = item.toJSON();
  expect(tree).toMatchSnapshot();
});
