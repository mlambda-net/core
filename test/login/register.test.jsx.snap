// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Test 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation10 MuiPaper-rounded"
>
  <div
    className="MuiBox-root MuiBox-root-5"
  >
    <div
      className="MuiBox-root MuiBox-root-6"
    >
      <h5
        className="MuiTypography-root MuiTypography-h4 MuiTypography-colorPrimary MuiTypography-gutterBottom"
      >
        Sign up
      </h5>
    </div>
    <div
      className="MuiBox-root MuiBox-root-7"
    >
      <div
        className="MuiBox-root MuiBox-root-8"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-error Mui-error"
              data-shrink={false}
              htmlFor="email"
              id="email-label"
            >
              Email
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={true}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="email"
              />
            </div>
            
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-9"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-error Mui-error"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
            >
              Password
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={true}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
            </div>
            
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-10"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-error Mui-error"
              data-shrink={false}
              htmlFor="confirm_password"
              id="confirm_password-label"
            >
              Confirm password
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={true}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="confirm_password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-11"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Sign up
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
